app/services/foreman_resource_quota/resource_origin.rb

```diff
@@ -1,69 +1,96 @@
-              attribute_content)
-          end
-          missing_res_per_host.delete(host_id) if missing_res_per_host[host_id].empty?
-        end
-      end
+      relevant_hosts = load_hosts_eagerly(missing_hosts_resources, host_objects, host_attribute_eager_name)
+      host_values = collect_attribute_from_hosts(relevant_hosts, host_attribute_name)
+      sum_resources_and_delete_missing_hosts!(resources_sum, missing_hosts_resources, host_values)
+    end
 
-      def host_eager_name
-        raise NotImplementedError
-      end
+    def host_attribute_eager_name
+      raise NotImplementedError
+    end
 
-      def host_attribute_name
-        raise NotImplementedError
-      end
+    def host_attribute_name
+      raise NotImplementedError
+    end
 
-      def extract_cpu_cores(param)
-        raise NotImplementedError
-      end
+    def extract_cpu_cores(param)
+      raise NotImplementedError
+    end
 
-      def extract_memory_mb(param)
-        raise NotImplementedError
-      end
+    def extract_memory_mb(param)
+      raise NotImplementedError
+    end
 
-      def extract_disk_gb(param)
-        raise NotImplementedError
-      end
+    def extract_disk_gb(param)
+      raise NotImplementedError
+    end
 
-      private
+    private
 
-      def collect_attribute_from_hosts(host_list, attribute_name)
-        host_values = []
-        host_list.each do |host|
-          host_attribute = host.send(attribute_name)
-          host_values << [host.id, host_attribute] if host_attribute.present?
-        rescue StandardError
-          # skip hosts whose attribute couldn't be collected
+    def load_hosts_eagerly(missing_hosts_resources, host_objects, eager_attribute)
+      relevant_hosts = Host::Managed.where(name: missing_hosts_resources.keys).includes(eager_attribute)
+      relevant_hosts = relevant_hosts.compact
+      if relevant_hosts.size < missing_hosts_resources.size # Add non-eagerly loaded host objects
+        relevant_hosts_names = relevant_hosts.map(&:name)
+        (missing_hosts_resources.keys - relevant_hosts_names).each do |missing_host_name|
+          relevant_hosts << host_objects[missing_host_name]
         end
-        host_values
       end
+      relevant_hosts
+    rescue ActiveRecord::AssociationNotFoundError
+      # the eager_attribute could not be loaded
+      host_objects.values
+    end
 
-      def process_resource!(resources_sum, missing_res_per_host, resource_name, host_id, attribute_content)
-        resource_value = method(RESOURCE_FUNCTION_MAP[resource_name]).call(attribute_content)
-        return unless resource_value
+    def collect_attribute_from_hosts(host_list, attribute_name)
+      host_values = {}
+      host_list.each do |host|
+        attribute_value = host.send(attribute_name)
+        host_values[host.name] = attribute_value if attribute_value.present?
+      rescue StandardError
+        # skip hosts whose attribute couldn't be collected
```

Comment:
What happens if you skip it? Any consequences?

Reply:
The error could occur because the host might not have the `attribute_name` attribute. So, line 59 `host.send(attribute_name)` could fail and this error is caught here.
Skipping this error without consequences is wanted behavior, because that means the host will be kept in the "missing_res_per_host" list. The next ResourceOrigin will try a different `attribute_name` which might work for this specific host then.